Starting a new project.md

```diff
@@ -0,0 +1,68 @@
+## Starting a new project
+
+When you're starting a new project, there are a few things you need to do:
+
+#### 1. Open a repository
+
+Ask your team lead or DevOps to open a repository.
+
+#### 2. Fill out Project setup sheet
+
+We've come up with a project setup template in cooperation with DevOps team. It's purpose is to provide the DevOps with all the details needed for deploying your application to staging and production servers.
+
+It contains general info about your project that your project manager can fill out as well as more technical information like frameworks and technologies used, build and deploy setup, third party services that need to be set up...
+
+Ask your PM for a link to the sheet template.
+
+**Fill in the blanks on Backend related stuff.**
+
+If you need any other services that aren't mentioned, **provide as many information as possible**
+
+**Project setup sheet needs to be up to date** with new changes to project setup and services used. If you're adding some new dependency or service or changing something, please update it in project setup sheet and notify DevOps to make these changes.
+
+
+#### 3. Generate new application
+
+Use [**default rails template**](https://github.com/infinum/default_rails_template/) to start your new application.
+
+Most of our work is based on the Rails framework. We have some standard configurations and gems that are basically in every Rails application. To make our lives easier, we've made a default template to generate new Rails applications that meet those standards.
+
+
+#### 4. Check for new/better gems
+
+We usually use specific gems for some purposes, e.g. Devise for authentication, Pundit for authorization... That doesn't mean we necessarily need to use those gems.
+
+When adding a gem to your application, check if there's some new and better gem for that purpose.
+
+When checking out a gem, always look for these things:
+* is the gem maintained? when was the last commit?
+* how many stars does it have?
+* what is the ratio between open and closed issues?
+
+
+#### 5. Fill out readme
+
+After setting up a project, fill out a readme file with all the necessary details about the application:
+* dependencies & prerequisites
+* setup instructions
+* development instructions
+* deployment instructions
+* test instructions
+* any other project specific information that someone who takes over a project needs to know
```

Comment:
The ER diagram should also be saved somewhere in the repository, maybe a separate file, and it should be linked from the readme.

Reply:
Does anybody use an ER diagram of an ongoing project?

It is either out of date (the one created before the coding started)
or it is created with railroady (or similar gem) and it can be a cluttered mess and unreadable.

I rather look at schema.db and follow the relalationships

Reply:
I think that keeping an ER diagram in sync with the schema creates overhead on ongoing projects because you have to remember to always update the diagram when you change the schema.

Also, [dbdiagram.io](https://dbdiagram.io) has an option to import schema.rb, so you can use that to quickly create a diagram if you'll ever need it:
<img width="225" alt="image" src="https://user-images.githubusercontent.com/36794420/85071935-8879bd80-b1b8-11ea-94f7-6fd53738f1a2.png">

Reply:
I like keeping the ERD in [repos](https://github.com/infinum/rails-messe-connect-networking#database-architecture) :)

It does sometimes get out of sync cause you forget or you just don't feel like updating it if you're adding a column or two...but every month or so it makes sense to me...

Reply:
I agree with @lovro-bikic. You'll always have to remember to keep it up to date and I think it's not efficient. If you're working on a project, you know it's architecture. And if you're new on the project you can check out schema and create a ER diagram with dbdiagram.io using schema.rb

Reply:
Maintaining the ERD diagram is a subjective and opinionated topic
* which tools to use
* rate of updates
* allowed diff between DB schema and ERD diagram
* artifact location...

We should discuss this topic internally, come to an agreement if it's necessary or not. If we agree to maintain ERD diagrams in projects, we'll also define the mechanics of how to do it. At that point, we can circle back and update the handbook.

The bottom line, we should move this discussion to a more appropriate channel.

Reply:
Ok. @nikajukic bring up this topic on the next backend talks

Reply:
@melcha deal! :)